src/AccountServices.js

```diff
@@ -0,0 +1,257 @@
+  log.info("Handler.AccountServices", `${event.sender} is requesting a twitter account link.`);
+  this._oauth_getUrl(event.sender).then( (url) =>{
+    intent.sendMessage(event.room_id, {
+      "body": `Go to ${url} to receive your PIN, and then type it in below.`,
+      "msgtype": "m.text"
+    });
+  }).catch(err => {
+    log.error("Handler.AccountServices", `Couldn't get authentication URL: ${err}` );
+    intent.sendMessage(event.room_id, {
+      "body": "We are unable to process your request at this time.",
+      "msgtype": "m.text"
+    });
+  });
+}
+
+/**
+ * _unlinkAccount - Processes a users request to unlink their twitter account
+ * from the bridge.
+ *
+ * @param  {object} event The Matrix Event from the requesting user.
+ */
+AccountServices.prototype._unlinkAccount = function (event) {
+  var intent = this._bridge.getIntent();
+  this._storage.remove_client_data(event.sender);
+  this._storage.remove_timeline_room(event.sender);
+  this._twitter.detach_user_stream(event.sender);
+  intent.sendMessage(event.room_id, {
+    "body": "Your account (if it was linked) is now unlinked from Matrix.",
+    "msgtype": "m.text"
+  });
+}
+
+/**
+ * _processPIN - The user has pasted a pin from the authorisation link we
+ *  provided.
+ *
+ * @param  {object} event The Matrix Event from the requesting user.
+ */
+AccountServices.prototype._processPIN = function (event) {
+  var intent = this._bridge.getIntent();
+  this._storage.get_twitter_account(event.sender).then((client_data) => {
+    var pin = event.content.body;
+    log.info("Handler.AccountServices", `${event.sender} sent a pin (${pin}) to auth with.`);
+    if(client_data == null) {
+      intent.sendMessage(event.room_id, {
+        "body": "You must request access with 'link account' first.",
+        "msgtype": "m.text"
+      });
+      return;
+    }
+    this._oauth_getAccessToken(pin, client_data, event.sender).then((profile) => {
+      intent.sendMessage(event.room_id, {
+        "body": "All good. You should now be able to use your Twitter account on Matrix.",
+        "msgtype": "m.text"
+      });
+      this._twitter.create_user_timeline(event.sender, profile);
+      this._twitter.attach_user_stream(event.sender);
+    }).catch(err => {
+      intent.sendMessage(event.room_id, {
+        "body": "We couldn't verify this PIN :(. Maybe you typed it wrong or you"
+             + "might need to request it again.",
+        "msgtype": "m.text"
+      });
+      log.error("Handler.AccountServices", "OAuth Access Token Failed:%s", err);
+    });
+  });
+}
+
+AccountServices.prototype._oauth_getAccessToken = function (pin, client_data, id) {
+  return new Promise((resolve, reject) => {
+    if (!client_data || client_data.oauth_token == "" || client_data.oauth_secret == "") {
+      reject("User has no associated token request data");
+      return;
+    }
+    this._oauth.getOAuthAccessToken(
+      client_data.oauth_token,
+      client_data.oauth_secret,
+      pin,
```

Comment:
Tempted to put this on the same line as `:174` and then just put the function on a new line, looots of vertical space otherwise.